homeassistant/components/enphase_envoy/sensor.py

```diff
@@ -283,25 +285,22 @@ def __init__(
         self.entity_description = description
         envoy_name = coordinator.name
         self._serial_number = serial_number
-        name = description.name
         key = description.key
 
         if key == INVERTERS_KEY:
             # Originally there was only one inverter sensor, so we don't want to
-            # break existing installations by changing the name or unique_id.
-            self._attr_name = f"{envoy_name} Inverter {serial_number}"
+            # break existing installations by changing the unique_id.
             self._attr_unique_id = serial_number
         else:
-            # Additional sensors have a name and unique_id that includes the
+            # Additional sensors have a unique_id that includes the
             # sensor key.
-            self._attr_name = f"{envoy_name} Inverter {serial_number} {name}"
             self._attr_unique_id = f"{serial_number}_{key}"
 
         envoy_serial_num = coordinator.envoy.serial_number
         assert envoy_serial_num is not None
         self._attr_device_info = DeviceInfo(
             identifiers={(DOMAIN, serial_number)},
-            name=f"Inverter {serial_number}",
+            name=f"{envoy_name} Inverter {serial_number}",
```

Comment:
```suggestion
            name=f"Inverter {serial_number}",
```

This is a change, and it seems like might be a bit too much

<img width="954" alt="Screenshot 2023-08-05 at 2 08 38 PM" src="https://github.com/home-assistant/core/assets/663432/1609a412-c384-4794-8b04-0440e78eace4">
<img width="679" alt="Screenshot 2023-08-05 at 2 09 26 PM" src="https://github.com/home-assistant/core/assets/663432/213de8bd-31bf-4c45-9b22-fa3054740e10">